src/server/game/Mails/Mail.cpp

```diff
@@ -219,43 +212,41 @@ void MailDraft::SendMailTo(CharacterDatabaseTransaction trans, MailReceiver cons
     else if (sender.GetMailMessageType() == MAIL_CREATURE && sBattlegroundMgr->GetBattleMasterBG(sender.GetSenderId()) != BATTLEGROUND_TYPE_NONE)
         expire_delay = DAY;
     // default case: expire time if COD 3 days, if no COD 30 days (or 90 days if sender is a game master)
+    else if (m_COD)
+        expire_delay = 3 * DAY;
+    else if (custom_expiration > 0)
+        expire_delay = custom_expiration * DAY;
     else
     {
-        if (m_COD)
-            expire_delay = 3 * DAY;
-        else if (custom_expiration > 0 )
-            expire_delay = custom_expiration * DAY;
-        else
-            expire_delay = pSender && pSender->GetSession()->GetSecurity() ? 90 * DAY : 30 * DAY;
+        Player* pSender = ObjectAccessor::FindPlayerByLowGUID(sender.GetSenderId());
+        expire_delay = pSender && pSender->GetSession()->GetSecurity() ? 90 * DAY : 30 * DAY;
     }
 
     time_t expire_time = deliver_time + expire_delay;
 
     // Add to DB
-    uint8 index = 0;
     CharacterDatabasePreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_INS_MAIL);
-    stmt->SetData(  index, mailId);
-    stmt->SetData (++index, uint8(sender.GetMailMessageType()));
-    stmt->SetData  (++index, int8(sender.GetStationery()));
-    stmt->SetData(++index, GetMailTemplateId());
-    stmt->SetData(++index, sender.GetSenderId());
-    stmt->SetData(++index, receiver.GetPlayerGUIDLow());
-    stmt->SetData(++index, GetSubject());
-    stmt->SetData(++index, GetBody());
-    stmt->SetData  (++index, !m_items.empty());
-    stmt->SetData(++index, uint32(expire_time));
-    stmt->SetData(++index, uint32(deliver_time));
-    stmt->SetData(++index, m_money);
-    stmt->SetData(++index, m_COD);
-    stmt->SetData (++index, uint8(checked));
```

Comment:
I prefer this as it is impossible to mess up if tables are added or removed.

Reply:
I have "reverted" it. I think your point is absolutely valid.